libshpool/src/config.rs

```diff
@@ -212,6 +237,37 @@ pub struct Config {
     pub motd_args: Option<Vec<String>>,
 }
 
+impl Config {
+    // Merge with `another` Config instance, with `self` taking higher priority.
+    //
+    // Top level options with simple value are directly taken from the higher priority instance.
+    // The merge strategy for list or map are handled case by case. Please refer to each options'
+    // documentation for details.
+    pub fn or(self, another: Config) -> Config {
+        Config {
+            norc: self.norc.or(another.norc),
+            noecho: self.noecho.or(another.noecho),
+            nosymlink_ssh_auth_sock: self
+                .nosymlink_ssh_auth_sock
+                .or(another.nosymlink_ssh_auth_sock),
+            noread_etc_environment: self.noread_etc_environment.or(another.noread_etc_environment),
+            shell: self.shell.or(another.shell),
+            // TODO: Check this
```

Comment:
dangling?

Reply:
I want to hear your opinion of the best strategy here. Should we do merge rather than overwrite for maps and vecs? What if the user want to add to the list/clear then add to the list/clear the list, etc?

Reply:
Can you remove these TODOs